request_handler.cc

```diff
@@ -73,6 +78,43 @@ Headers Request::headers() const {
 
 // Response Implementations
 
+std::unique_ptr<Response> Response::Parse(const std::string& raw_response) {
+    std::unique_ptr<Response> new_response(new Response());
+    std::vector<std::string> lines = split_lines(raw_response);
+
+    // parse first line of the response
+    if (lines.size() > 0) {
+        auto first_space = lines[0].find(" ");
+        std::string temp = lines[0].substr(first_space + 1);
+        auto second_space = temp.find(" ");
+        std::string status_code = temp.substr(0, second_space);
+        try {
+            new_response->status_code_ = std::stoi(status_code);
```

Comment:
Maybe use the function SetStatus(  instead of directly accessing the variable, although not necessary
Although, I am not even sure how you are able to access that since its a private member variable